INSTALLATION.md

```diff
@@ -35,25 +45,47 @@ Follow the setup guide for `git` on official [git docs](https://git-scm.com/down
 
-This will download a local copy of talawa-api in that directory.
+3. Navigate to [https://github.com/PalisadoesFoundation/talawa-api/](hhttps://github.com/PalisadoesFoundation/talawa-api/) and click on the fork button. It is placed on the right corner opposite the repository name `PalisadoesFoundation/talawa-api`.
 
-<br/>
+![Image with fork](./image/install1.png)
 
-## Change directory into the cloned repo
+4. You should now see `talawa-api` under your repositories. It will be marked as forked from `PalisadoesFoundation/talawa-api`
 
-Right after cloning the repo you can change the directory of your current `terminal(shell)` to the root directory of cloned repository using this command:-
+![Image of user's clone](./image/install2.png)
 
-        cd ./talawa-api
+5. Clone the repository to your local computer (replacing the values in `{{}}`):
 
-**NOTE:-** `All the commands we're going to execute in the following instructions will assume you are in the root directory of the project. If you fail to do so, the commands will not work.`
+```
+$ git clone https://github.com/{{GITHUB USERNAME}}/talawa-api.git
+```
 
-<br/>
+Now your origin remote is pointing to your fork (`{{GITHUB USERNAME}}/talawa-api`). To stay up to date with the main `PalisadoesFoundation/talawa-api` repository, add it as a remote called upstream. You'll first need to move into the `talawa-api` directory that was created by the clone operation.
+
+```
+$ cd talawa-api
+$ git remote add upstream https://github.com/PalisadoesFoundation/talawa-api.git
+$ git remote -v
+origin     https://github.com/{{GITHUB USERNAME}}/talawa-api.git (fetch)
+origin     https://github.com/{{GITHUB USERNAME}}/talawa-api (push)
+upstream   https://github.com/PalisadoesFoundation/talawa-api.git (fetch)
+upstream   https://github.com/PalisadoesFoundation/talawa-api.git (push)
+```
+
+The git remote -v command at the end shows all your current remotes.
+
+Now you can pull in changes from `PalisadoesFoundation/talawa-api` by running `git pull upstream {{branch}}` and push your changes to your fork by running `git push origin {{branch}}`.
+
+We have established a clean setup now. We can make any changes we like and push it to this forked repository, and then make a pull request for getting the changes merged into the original repository. Here's a nice picture explaining the process ([image source](https://github.com/Rafase282/My-FreeCodeCamp-Code/wiki/Lesson-Save-your-Code-Revisions-Forever-with-Git)).
+
+![Diagram of the fork-and-clone workflow](./image/install3.png)
+
+For making any changes to original repository, we first sync our cloned repository with original repository. We merge `develop` with `upstream/develop` to do this. Now we make a new branch, do the changes on the branch, push the branch to forked repository, and make a PR from Github interface. We use a different branch to make changes so that we can work on multiple issues while still having a clean version in develop branch.
```

Comment:
I don't think we need to change this part because it is very basic information and everyone has there own cycle  for pull and push, we should not complicate the process. I personally clone and pull from palisadoes directly to my local machine and push to my forked repo to avoid commit history conflicts. 

@palisadoes Your thoughts on this ?

Reply:
True, but our installation documentation has historically been poor. This is an area where greater clarity without additional complication is valuable. We need to eliminate barriers that prevent newcomers from wanting to contribute. I think this update is in the wrong file.

@EshaanAgg 
1. Some of this information should be moved to an **upgrading** section. Setting upstream would be a simple way of doing this followed by a git hard reset. For example:

            git fetch upstream
            git reset upstream/develop --hard

1. Interacting with both the upstream **AND** the origin should be put in the CONTRIBUTING.md file as these are common developer activities.setting the upstream is really for that purpose.

This INSTALLATION.md file should focus on getting the system up and running.

@noman2002 

Your strategy is for the CONTRIBUTING.md file. @EshaanAgg in that file we could recommend that contributors do a hard reset prior to working on issues so that they have the latest version of the code which will:

1. Mute the commit histories when submitting the PR 
2. Minimize the risk of unwanted files from previous work being accidentally submitted 

Please add CONTRIBUTING.md to this PR

Reply:
Got it. I'll migrate the documentation accordingly.